ports/rp2/mphalport.c

```diff
@@ -176,11 +176,22 @@ void mp_hal_delay_ms(mp_uint_t ms) {
     }
 }
 
-uint64_t mp_hal_time_ns(void) {
+static uint64_t time_ns_offset;
+
+void mp_hal_time_ns_set_from_rtc(void) {
+    // Delay at least one RTC clock cycle so it's registers have updated with the most
+    // recent time settings.
+    sleep_us(23);
+
     datetime_t t;
     rtc_get_datetime(&t);
+    uint64_t us = time_us_64();
```

Comment:
Reorder this with the line below it (you compute `s` from `t`, so do that before computing `us`).

Reply:
I want to sample the RTC and time_us_64() as close together as possible, to make sure they match as closely as possible at future times.  I can add a comment to this effect.

Reply:
Comment added.